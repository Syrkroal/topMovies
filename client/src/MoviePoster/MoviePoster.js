import "./MoviePoster.css";
import { useHistory } from "react-router-dom";

const imgPath = "https://image.tmdb.org/t/p/w500/";

function MoviePoster(props) {
    let history = useHistory();
    const movie = props.movie;

    function gotoDetails() {
        history.push("/details", { movie: movie });
    }

    return (
        <div className="movie-container" onClick={gotoDetails}>
            <img
                className="movieImg"
                src={imgPath + movie.poster_path}
                alt="Poster"
            />
            <div className="movie-title">{movie.title}</div>
        </div>
    );
}

export default MoviePoster;
