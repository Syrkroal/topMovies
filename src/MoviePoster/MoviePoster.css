.movieImg {
    height: 300px;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
}

.movie-container {
    border-radius: 10px;
    background-color: #141414;
    cursor: pointer;
    margin: 20px;
    width: 200px;
    display: flex;
    flex-direction: column;
    color: white;
    font-size: 20px;
}

.movie-title {
    display: flex;
    justify-content: center;
    font-size: 16px;
    margin: 10px;
}
