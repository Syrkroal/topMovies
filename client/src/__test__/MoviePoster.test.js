import React from "react";
import Enzyme, { shallow } from "enzyme";
import MoviePoster from "../MoviePoster/MoviePoster";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("MoviePoster to match snapshot", () => {
    const movie = {
        poster_path: "path",
        title: "title",
    };
    const container = shallow(<MoviePoster movie={movie} />);
    it("should match the snapshot", () => {
        expect(container.html()).toMatchSnapshot();
    });
});
