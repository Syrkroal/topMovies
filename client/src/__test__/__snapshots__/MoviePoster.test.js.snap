// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster to match snapshot should match the snapshot 1`] = `"<div class=\\"movie-container\\"><img class=\\"movieImg\\" src=\\"https://image.tmdb.org/t/p/w500/path\\" alt=\\"Poster\\"/><div class=\\"movie-title\\">title</div></div>"`;
